Initial updates to document o11y support in Anbox Cloud (#280)
canonical/anbox-cloud-docs: howto/monitor/landing.md

(howto-monitor-anbox)=
# How to monitor Anbox Cloud

Anbox Cloud collects various metrics and makes them accessible through API endpoints. While Anbox Cloud does not provide its own observability solution, it supports integrating with external solutions.

You can find a list of available metrics at {ref}`ref-prometheus-metrics`.

## Access metrics with the appliance

The Anbox Cloud Appliance provides a central metrics endpoint which aggregates metrics from all internal services and Anbox instances.

If you haven't enabled metrics support at initialization time, you can do that now by running

    sudo anbox-cloud-appliance enable metrics

If you want to disable metrics again you can run

    sudo anbox-cloud-appliance disable metrics

Afterwards you can find the URL which you can use to retrieve the metrics in the [Prometheus data format](https://prometheus.io/docs/concepts/data_model/).

    sudo anbox-cloud-appliance config show

> The output will list a `metrics.url` field which provides you the URL. The output will also provide a TLS certificate the endpoint the URL points to is using to establish a secure and authenticated connection.

Reviewer: I think there are some superfluous words here after 'TLS certificate', but I'm not sure what part you want to keep.

You can either configure a Prometheus instance to scrape the endpoint or manually retrieve the metrics via `curl`, for example

  metrics_url="$(sudo anbox-cloud-appliance config show | yq .metrics.url)"

>   sudo anbox-cloud-appliance config show | yq .metrics.tls.certificate > metrics.pem

Reviewer: yq is not that well-known compared to jq, so I'm not sure if that will not bring questions from users.

Author: You're right, let me add a hint about `yq`.

(howto-monitor-anbox)=
# How to monitor Anbox Cloud

Anbox Cloud collects various metrics and makes them accessible through API endpoints. While Anbox Cloud does not provide its own observability solution, it supports integrating with external solutions.

You can find a list of available metrics at {ref}`ref-prometheus-metrics`.

## Access metrics with the appliance

The Anbox Cloud Appliance provides a central metrics endpoint which aggregates metrics from all internal services and Anbox instances.

If you haven't enabled metrics support at initialization time, you can do that now by running

    sudo anbox-cloud-appliance enable metrics

If you want to disable metrics again you can run

    sudo anbox-cloud-appliance disable metrics

> Afterwards you can find the URL which you can use to retrieve the metrics in the [Prometheus data format](https://prometheus.io/docs/concepts/data_model/).

Reviewer 2: ```suggestion
To retrieve the metrics in the [Prometheus data format](https://prometheus.io/docs/concepts/data_model/), run:
```

    sudo anbox-cloud-appliance config show

> The output will list a `metrics.url` field which provides you the URL. The output will also provide the TLS certificate of the the endpoint the URL points to. This should be used to establish a secure and authenticated connection.

Reviewer 2: ```suggestion
The output will list a `metrics.url` along with the TLS certificate of the endpoint referred by the URL. This certificate should be used to establish a secure and authenticated connection.
```

You can either configure a Prometheus instance to scrape the endpoint or manually retrieve the metrics via `curl`, for example

>     # We need q in order to parse and process the YAML output

Reviewer: Typo:
```suggestion
    # We need yq in order to parse and process the YAML output
```

>     sudo apt install  -y yq jq

Reviewer: `yq` is not available as a deb, only as a snap.

Author: It's available as a deb starting with noble, see https://packages.ubuntu.com/noble/yq

    metrics_url="$(sudo anbox-cloud-appliance config show | yq .metrics.url)"
    sudo anbox-cloud-appliance config show | yq .metrics.tls.certificate > metrics.pem
    curl --cacert ./metrics.pem "$metrics_url"

> As output you will see all available metrics, include metrics for the individual Anbox instances.

Reviewer 2: ```suggestion
You will see all available metrics as output, including metrics for the individual Anbox instances.
```
